### DISCUSSION TITLE
Und was wenn ich keine senec hab?

### DISCUSSION THREAD

**Discussion by tsfuchs** [solectrus/solectrus#1620] [General]
Hallo zusammen,
Ich suche nach einem Dashboard das mir Erzeugung eines balkonkraftwerks und Gesamtverbrauch im Haus gegenüberstellt. Dabei bin ich über evcc (das eigentlich wallboxen auswertet) hier gelandet.
(Leider) habe ich keine senec Anlage, aber die nötigen Daten landen über iobroker per mqtt bzw. shelly Adapter in einer influx Datenbank. Wäre es irgendwie möglich das Dashboard daran anzuzapfen?

**Comment by ledermann** [comment]
Grundsätzlich ist SOLECTRUS so aufgebaut, dass es nicht zwingend nur mit einer SENEC-Anlage funktioniert. Woher die Messwerte stammen (also Inverter-Leistung, Hausverbrauch, Wallbox-Leistung etc.), ist der Dashboard-App recht egal. 

Für die Abfrage der Messwerte aus SENEC.Home gibt es den von mir entwickelten [SENEC collector](https://github.com/solectrus/senec-collector), der wirklich SENEC-spezifisch ist. Entsprechend wäre es denkbar, dass jemand mit Programmierkenntnissen einen Collector baut, der ein anderes Gerät ansteuert. In welcher Programmiersprache das geschieht, ist recht egal - Hauptsache, die Messwerte landen dann im passenden Format in InfluxDB. Ich selbst kann das jedoch leider nicht erledigen, da ich nur eine SENEC-Anlage zur Verfügung habe. In einer idealen Welt würde es also etliche Collectoren geben, um Photovoltaik-Messwerte von Geräten unterschiedlichster Hersteller abzufragen und an die InfluxDB von SOLECTRUS zu übermitteln.

@tsfuchs Deinen Vorschlag, dass SOLECTRUS auf eine **andere**, also fremde InfluxDB zugreift, finde ich etwas problematisch - denn auch, wenn da ähnliche Werte drinstehen, sind die sicherlich auf andere Art strukturiert und bezeichnet. Hier könnte man vielleicht ein Mapping vornehmen, aber das klingt für mich nicht nach einer sauberen Lösung. Ich strebe das daher selbst nicht an. Aber hey, das ist alles Open-Source hier - ich freue mich über jeden, der sich in die Weiterentwicklung einbringen möchte :-)

**Reply by GrimmiMeloni** [reply]
Verständlich. Ich würde mir das zumindest für meine Anlage (Tesla Powerwall + Easee WB + SolarEdge WR) einmal näher anschauen. Gibt es irgendwo eine Aufstellung der Werte die in Influx erwartet werden?

**Reply by ledermann** [reply]
Derzeitig ist die Struktur so (Bucket/Measurement/Fields):

```
Bucket: SENEC
  Measurement: SENEC
    - inverter_power (Stromerzeugung in W, Float)
    - grid_power_minus (Netz-Einspeisung in W, Float)
    - grid_power_plus (Netz-Entnahme in W, Float)
    - house_power  (Hausverbrauch in W, Float)
    - wallbox_charge_power (Wallbox-Leistung in W, Float)
    - bat_power_minus (Akku-Entnahme in W, Float)  
    - bat_power_plus (Akku-Beladung in W, Float)  
    - bat_fuel_charge (Batterie-Ladestand in %, Float)  
    - case_temp (Gehäuse-Temperatur in °C, Float)
    - current_state (Aktueller Status, String)
    - mpp1_power (String 1 Stromerzeugung in W, Float, derzeitig noch nicht verwendet)
    - mpp2_power (String 2 Stromerzeugung in W, Float, derzeitig noch nicht verwendet) 
    - mpp3_power (String 3 Stromerzeugung in W, Float, derzeitig noch nicht verwendet)
    - bat_charge_current (Batterie-Ladestrom in A, Float, nicht benötigt)
    - bat_voltage (Batterie-Lade-Spannung in V, Float, nicht benötigt)
```

Der Bucket-Name ist anpassbar via `.env`, der Measurement-Name derzeitig aber nicht, was unschön ist.
Ich bin aber gerne bereit, die Struktur anzupassen, zu verbessern und Dir weitere Informationen zukommen zu lassen. Wäre toll, wenn dabei eine SOLECTRUS-Anbindung für andere Gerätschaften herauskommt :-)

**Reply by GrimmiMeloni** [reply]
Danke für die Info. 

Da ich die Daten bereits in Influx habe, könnte über eine Continuous Query oder ggf. über Kapacitor ein mapping auf die erwarteten Fields erfolgen. Der Nachteil hierbei wäre, daß es sich um eine spezifische Lösung für mein Setup handelt. Außerdem fehlen mir noch die Daten von der WB. Fazit: Geht, aber nur bedingt brauchbar.

Interessanter ist es, auf mein bestehendes `evcc` setup zu setzen. `evcc` hat bereits "Adapter" für unzählige Hardwarekomponenten, und stellt die Messwerte u.a. per REST API, aber auch MQTT bereit. Hier könnte man ansetzen um die Daten zu beziehen. Das könnte in einer externen Komponente (Container der von MQTT liest und in Influx schreibt) erfolgen. Dann hätte man einen Adapter zwischen evcc und Solectrus, der auch von anderen verwendet werden könnte.

**Reply by GrimmiMeloni** [reply]
@ledermann ich bin heute bereits ein ganzes Stück weiter gekommen und habe jetzt eine initiale Anbindung an evcc wie oben skizziert gebaut. Diese ersetzt im docker-compose bei mir aktuell den Senec scraper. Funktioniert erstmal rudimentär.

Noch einige Fragen:
- Was ist case_temp genau für eine Temperatur? Vom Wechselrichter?
- Welche Werte sind in current_state vorgesehen? Und wann werden diese gesetzt?
- In der UI sehe ich im Wechsel mal ein "FAIL" und mal ein Live oben rechts. Was genau wird damit abgebildet? Wann erscheint hier FAIL?

**Reply by ledermann** [reply]
Wow, das hört sich wirklich gut an! Wenn ich es richtig verstehe, greifst Du nun per MQTT auf evcc zu, korrekt? Kann man Deine Arbeit einsehen, gibt es ein öffentliches Repository?

Vielleicht hast Du es schon gesehen: Es gibt noch eine weitere Initiative, Daten per MQTT abzugreifen:
https://github.com/loebse/mqtt-collector

Zu Deinen Fragen:

1) `case_temp` ist die Gehäuse-Temperatur, wie sie vom SENEC-Speicher gemeldet wird. Dieser liefert noch weitere Temperaturwerte, z.B. für jeden Akku separat. Der Einfachheit halber habe ich den einen genommen. Bei einem anderen Speicher oder Wechselrichter gibt es vielleicht einen Wert, der aussagekräftig ist. Denkbar wäre auch, die Raumtemperatur zu nehmen, sofern vorhanden. Es geht um eine Temperatur, bei der man sehen kann, ob alles im grünen Bereich liegt.

2) In `current_state` können beliebige (kurze) Strings stehen, die den aktuellen Status beschreiben und mit jeder einzelnen Messung vom SENEC-Collector geliefert werden. Andere Geräte werden vermutlich auch einen Status-Wert haben, den man hier liefern könnte. Falls nicht, kann er leer bleiben. Ich habe absichtlich keine Integer-ID (mit einem Mapping auf Strings) benutzt, um möglichst flexibel zu sein. Den erhöhten Speicherplatzbedarf durch Speicherung von Strings habe ich in Kauf genommen.

3) Im Dashboard wird "FAIL" angezeigt, wenn die letzte Messung aus InfluxDB mehr als **10 Sekunden** her ist. Andernfalls, also im Normalfall, wird der Wert aus `current_state` angezeigt bzw. "LIVE", wenn es keinen Status-String gibt. Die 10 Sekunden sind derzeitig konstant, es gibt aber die Idee, das einstellbar zu machen (siehe #1919). Um den Live-Charakter des Dashboards zu erhalten, empfehle ich aber, Messwerte im kurzen Abstand zu schicken. Der SENEC-Collector liefert standardmäßig im 5-Sekunden-Abstand.

**Reply by GrimmiMeloni** [reply]
Ja korrekt. Ich lese jetzt per MQTT die Werte von evcc. 
Das Projekt von @loebse kannte ich noch nicht, es macht aber ziemlich exakt das gleiche.
Der einzige Unterschied den ich auf die schnelle erkenne, ist das Handling der positiven und negativen Werte - bei evcc ist es genau anders herum.

Mein Projekt ist aktuell noch private, da es noch komplett zusammengehackt ist, inkl. hardcodierter tokens. Da ich auch einen Container benötige, könnte der einfachere Weg eine Contribution zum o.g. Projekt sein. Dort ist man schon weiter, und die Anpassung der positiv/negativ Logik dürfte kein Hexenwerk sein. Ich denke das schaue ich mir nochmal genauer an.

**Reply by GrimmiMeloni** [reply]
Habe gerade https://github.com/loebse/mqtt-collector/pull/13 geöffnet. Damit funktioniert es hier jetzt auch.

**Reply by ledermann** [reply]
Klingt gut! Das heißt, bei Dir läuft nun erfolgreich SOLECTRUS und die Messwerte kommen aus evcc? Läuft das zuverlässig, ohne Probleme?

Ich werde zu dem Projekt von @loebse auch noch ein paar PRs schicken, um das Ganze etwas aufzuhübschen (insbesondere Tests ergänzen).

**Reply by GrimmiMeloni** [reply]
Es sieht erstmal stabil aus, ja. Ich beobachte das jetzt erstmal ein paar Tage.

Ich habe für den mqttcollector eine Beispiel .env für die Anbindung an evcc beigesteuert. (siehe https://github.com/loebse/mqtt-collector/blob/main/.env.evcc.example) 

Es waren noch Fehler beim mapping der (negativen) Werte von evcc auf solectrus drin, das habe ich in https://github.com/loebse/mqtt-collector/pull/15 behoben. Für early access kann mein Fork verwendet werden, bis @loebse nochmal merged.

Im docker-compose habe ich den mqttcollector jetzt statt des Senec Collectors wie folgt drin:
```
  mqttcollector:
    image: ghcr.io/loebse/mqtt-collector:main
    container_name: mqttcollector_solectrus
    hostname: mqttcollector
    restart: always
    depends_on:
      - influxdb
    env_file:
      - mqtt-collector.env
```

Das genannte image ist noch nicht aktualisiert, ich teste hier mit einem lokalen Build. Sobald der o.g. merge durch ist, und @loebse nochmal ein publish des docker images durchgeführt hat sollte das aber wie gezeigt funktionieren. Die Datei mqtt-collector.env ist eine angepasste Kopie des .env aus dem mqttcollector repo.

**Reply by loebse** [reply]
Ist gemerged @GrimmiMeloni 

Freut mich, dass ihr mit am MQTT-Collector arbeitet. Ich habe hoffentlich auch bald wieder ein bisschen Zeit dafür.

**Reply by GrimmiMeloni** [reply]
@ledermann ich glaube hier gibt es noch ein Problem. Ich sehe häufiger mal "hängende" Werte. Insbesondere bei der Akkube- und Entladung. Bisheriges Highlight war der Zustand das der Akku gleichzeitig be- und entladen wurde. 🤡

Ich _vermute_ das es sich dabei um den jeweils letzten Wert in Battery Plus und Minus aus der Influx handelt.

Frage: Was ist die Erwartung beim Schreiben der Measurements? Müssen immer alle Fields geschrieben werden? Also insbesondere die Frage, muß z.B. wenn BattryPowerPlus geschrieben wird, auch immer ein Wert 0 in BatteryPowerMinus geschrieben werden? Aktuell schreiben wir keine expliziten Nullwerte, und folglich bleibt immer eines der beiden Fields leer (bzw. fehlt). Das selbe gilt auch für Grid.

**Reply by ledermann** [reply]
Ich habe zwischenzeitlich auch länger mit dem MQTT-Collector experimentiert und bin auf diverse Problemchen gestoßen.

Ich muss ein wenig ausholen: Der SENEC-Collector arbeitet so, dass alle 5 Sekunden über **einen** http-Request ein **Paket** an Messwerten vom SENEC-Speicher abgeholt wird. Das Paket enthält die für SOLECTRUS relevanten Messwerte (derzeitig 11) sowie einen Zeitstempel `{"STATISTIC": "MEASURE_TIME"}`, der präzise angibt, wann genau die Messwerte ermittelt wurden. Dieser Zeitstempel stammt direkt vom SENEC-Speicher (der seine interne Uhr per ntp aktuell hält). Das gesamte Datenpaket inklusive Zeitstempel wird dann in einem Rutsch an InfluxDB übertragen. Intern arbeitet der SENEC-Collector mit einer Queue, in der die Pakete so lange liegen, bis sie erfolgreich bei InfluxDB abgeliefert wurden (nützlich, falls z.B. InfluxDB mal down ist). Somit erfährt InfluxDB auch bei verzögerter Übertragung immer den korrekten Zeitstempel einer Messung.

Beim MQTT-Collector liegen die Dinge etwas anders: Hier obliegt das Abholen der Messwerte bei ioBroker, evcc oder sonstigem Tool. Bei ioBroker (über den Adapter https://github.com/nobl/ioBroker.senec) ist es so, dass dieser hintereinander ~jeweils **einzelne** Messwerte vom SENEC-Speicher abholt. Dadurch wird mir auch klar, warum es in den Docs die Warnung gibt, das Pull-Intervall nicht zu klein zu wählen, um den SENEC-Speicher nicht zu überlasten: Der SENEC-Adapter von ioBroker bombardiert den Speicher mit vielen Einzel-Abfragen innerhalb kürzester Zeit.~ (Korrektur: Stimmt nicht, siehe weiter unten) Zur Abfederung des Problems gibt es eine Unterscheidung zwischen Messwerten mit hoher und niedriger Priorität, sodass einige Werte deutlich öfter abgeholt werden als andere. Der Zeitstempel wird aber **niemals** abgeholt. Der MQTT-Broker erfährt somit **nicht**, von wann genau die Messwerte überhaupt stammen. Bei evcc (habe ich mir nur kurz angesehen) ist es ähnlich, auch dort gibt es mehrere Requests und der Zeitstempel wird **nicht** abgeholt.

Daraus ergeben sich relevante Unterschiede zwischen den beiden Collectoren:

- Da der SENEC-Collector paket-orientiert arbeitet, kann es nicht zu inkonsistenten Messwerten kommen. Es gibt also z.B. immer nur Ladung **oder** Entladung der Batterie. Der jeweils andere Wert ist 0. Beim MQTT-Collector kann es aber so aussehen, als würde die Batterie gleichzeitig laden und entladen. Das ist natürlich nur ein Artefakt der unterschiedlichen Abfrage-Zeitpunkte. Abhilfe könnte hier sein, wie von Dir vorgeschlagen, den jeweils anderen Wert (für Battery und Grid) explizit auf 0 zu setzen. Für alles andere gilt: Prinzipiell müssen nicht immer alle Werte geschrieben werden, InfluxDB sollte auch mit Lücken umgehen können und nimmt jeweils den letzten vorliegenden Wert.

- Die Zeitstempel der Messwerte sind beim MQTT-Collector grundsätzlich nicht korrekt bzw. nicht präzise. Sie stammen vom Zeitpunkt, an dem eine MQTT-Message beim Client (also dem MQTT-Collector) **ankommt**. Das ist nicht der Zeitpunkt der Messung und auch nicht der Zeitpunkt, zu dem der MQTT-Broker den Messwert **erhalten** hat. Ich kann noch nicht recht absehen, welche Auswirkungen das hat. In jedem Fall sorgt das für Abweichungen, die sich in SOLECTRUS insbesondere bei Zeitraum-Betrachtungen aufsummieren könnten. Auch fällt auf, dass SOLECTRUS häufig ein FAIL anzeigt, weil der letzte Messwert zu alt sein soll (also älter als 10 Sekunden) - mir ist noch nicht ganz klar, wie das entsteht.

Soweit erst mal der Stand der Dinge. Speziell bezüglich der Zeitstempel muss ich noch weiter recherchieren. Auch muss ich noch genauer prüfen, ob SOLECTRUS bei Lücken in den Messwerten immer korrekt arbeitet.

**Reply by GrimmiMeloni** [reply]
Danke für die Bestätigung und Erklärungen.

Das mit dem Zeitstempel ist vielleicht gar nicht so entscheidend. Die Messwerte kommen zumindest bei EVCC immer regelmäßig en-bloc. Auch wenn ein einziges Measurement mit allen Feldern "korrekter" ist, so sollte eine kurze Verzögerung zwischen den einzelnen Nachrichten noch im Toleranz Bereich liegen, oder nicht? Im Endeffekt "bastelt" die Influx Query dann einen Satz zusammen. Hier wäre noch interessant welcher Zeitstempel gewählt wird.
Wenn es ganz hart auf hart kommt, könnte man natürlich auch im Collector MQTT Nachrichten über ein Sekunden langes Zeitfenster einsammeln, und dann immer die letzten Werte aus dem Zeitfenster als ein Measurement in Influx schreiben. Eine 100% Garantie das alle Messwerte kommen hat man natürlich nicht (hängt vom Timing ab, und wie der Sender arbeitet) - bei evcc zumindest ist die Intervallänge in der **alle** Werte kommen aber bekannt. Da müßte sich was solides bauen lassen. Und schlimmer als das alle Werte als einzelnes Measurement geschrieben werden (so wie jetzt aktuell) wird es dadurch ja auch definitiv nicht.

Bzgl. der 0 Werte. Das können wir natürlich auch noch einbauen. Allerdings bewegen wir dann den MQTT-Collector immer weiter weg von einer "neutralen" Komponente, die einfach MQTT Nachrichten in Influx schreibt. Die 0-Werte zu faken wäre jetzt sicherlich nochmal Solectrus spezifischer. Andererseits sind die Felder in der Config auch Solectrus spezifisch. Vielleicht also auch kein Problem. Ich würde hier lokal mal das Schreiben der 0-Werte einbauen. @loebse hast du irgendwelche Einwände oder andere Vorschläge?

**Reply by loebse** [reply]
Also das Thema 0-Werte schreiben sehe ich unproblematisch, das kannst du gerne umsetzen @GrimmiMeloni. Notfalls würde sich das ja auch noch konfigurierbar machen lassen, wenn nötig.

Das Thema Einzelwerte va. "en bloc" sehe ich schon als wesentlich diffiziler an. Wenn ich die ioBroker Senec Doku richtig verstehe, wurde das gemacht, um einzelne Felder häufiger zu lesen als andere und um eine mögliche "Überlastung" zu verhindern @ledermann. Wobei sich mir diese Logik nicht ganz  erschließt. Sind mehrere kurze Anfragen wirklich besser als eine große Anfrage?

Im ioBroker selbst gibt es zumindest "Timestamp" und "Last changed" für jeden Wert. Also wann gab es das letzte Update und wann wurde der Wert zuletzt tatsächlich geändert. Allerdings sehe ich ad-hoc keine Möglichkeit, das über den MQTT Adapter zu veröffentlichen.

**Reply by GrimmiMeloni** [reply]
Das Schreiben von 0-Werten ist jetzt (optional) eingebaut. PR ist offen. Ich hatte das seit Sonntag lokal laufen, und sieht gut aus. Ich habe in den vergangenen Tagen keine Unstimmigkeiten mehr auf dem Solectrus Dashboard gesehen.

@ledermann ich habe noch eine Frage bzgl. der Messwerte. Wie granular (0,xxx kW) sind die Werte die ein Senec System ausgibt? Hintergrund meiner Frage ist, daß Tesla das mit Auflösung bis auf Watt ausgibt. Das bietet aber keinen echten Mehrwert, und gleichzeitig erzeugt es relativ viel "Lärm" im Dashboard, insbesondere wenn um den 0-Punkt herum Werte schwanken. Sowohl in den Graphen gibt es dann "zitternde" Linien, aber auch im Echtzeit-Balkendiagramm ist es eher irritierend wenn immer wieder z.B. ein 1-Pixel breiter roter Streifen für Netzbezug angezeigt wird. Ist das beim Senec genauso?

**Reply by ledermann** [reply]
@GrimmiMeloni Ja, grundsätzlich dürfte es egal sein, ob die Messungen zusammenfasst in einem Block an InfluxDB übertragen werden oder separat mit jeweils abweichendem Zeitstempel. InfluxDB wird das bei den Abfragen dann richtig berücksichtigen.

Beim Zeitstempel habe ich nur die Befürchtung, dass der auch mal grob daneben liegen kann, wenn nämlich Nachrichten - aus was für Gründen auch immer - später beim MQTT-Collector ankommen. Wie @loebse schon geschrieben hat, kennt zwar ioBroker den Zeitstempel (auch hier: nicht der Messung, sondern des Eingangs beim ioBroker), schickt ihn aber nicht per MQTT weiter. Insofern wird der native SENEC-Collector immer genauer sein als ein MQTT-Collector. Möglicherweise sind die Abweichungen aber vernachlässigbar klein, das müsse ich mal genauer vergleichen.

Ich denke nicht, dass es notwendig ist, Nachrichten erst zu sammeln und dann als Paket weiterzuleiten. Das macht die Sache vermutlich eher schlimmer, weil dann SOLECTRUS nervös wird und FAIL anzeigt :-) 

Dass jetzt die 0-Werte gesetzt werden, finde ich prima. Dass es bei SOLECTRUS überhaupt die Trennung von Grid und Batterie in plus/minus gibt, hat seinen Ursprung in den vom SENEC-Speicher gelieferten Daten. Die kommen nämlich genau so und das habe ich übernommen. Aus jetzigen Sicht hätte man das auch anders machen können, aber nun verlangt SOLECTRUS diese Struktur.

@loebse Warum der ioBroker (konkreter, der zugehörige [SENEC-Adapter](https://github.com/nobl/ioBroker.senec)) zwischen Werten niedriger und hoher Priorität unterscheidet, hat wohl folgenden Grund: ~Die Werte werden alle EINZELN vom Speicher abgeholt, d.h. jeweils mit einem eigenen POST-Request. Und das sind dann ziemlich viele. Um den Speicher nicht in die Knie zu zwingen, wird eben versucht, diejenigen Werte, die sich voraussichtlich häufiger ändern, auch häufiger abzurufen und die anderen dann eher seltener. Was ich aber überhaupt nicht verstehe ist, warum die Werte nicht über EINEN einzigen Request gesammelt abgeholt werden. Das geht nämlich auch und macht der SENEC-Collector so. Ich würde mal vermuten, dass das gesammelte Abholen weit weniger Last auf dem Speicher erzeugen würde.~ Stimmt nicht, siehe nächster Kommentar von mir.

@GrimmiMeloni Der SENEC-Speicher liefert Leistungswerte in Watt als Float mit einer Nachkommastelle. Der SENEC-Collector rundet das (weil ich das für eine unnötig hohe Genauigkeit halte) auf ganze Watt und legt das dann als Integer in der InfluxDB ab. Einige Werte wie Temperatur und Ladezustand werden aber als Float mit einer Nachkommastelle belassen. Dass sehr kleine Werte Darstellungsprobleme bereiten, ist mir aber noch nicht aufgefallen. Ja, es kann sein, dass links der Balken "Herkunft" einen sehr kleinen, 1px hohen Anteil für den Netzbezug anzeigt. Das ist mir aber bislang nicht als störend aufgefallen. 
Ansonsten versuche ich in SOLECTRUS bereits, an bestimmten Stellen sehr kleine Werte speziell zu behandeln. So wird beispielsweise der Badge mit der Verbrauchsquote erst dann angezeigt, wenn mindestens 50 Watt reinkommen, sodass es hier kein Flackern gibt. Dieses Prinzip könnte ich auch an anderen Stellen anwenden.

Ich habe in den nächsten Tagen wieder mehr Zeit für SOLECTRUS. Dann werde ich mir den aktuellen Stand des MQTT-Collector im Zusammenspiel mit ioBroker und evcc nochmal genauer ansehen :-)

**Reply by ledermann** [reply]
@loebse Ich muss mich korrigieren, was meine obige Aussage zum ioBroker-SENEC-Adapter betrifft: Es ist **nicht** so, dass dieser Adapter alle Werte einzeln vom Speicher abholt. Im [Quelltext](https://github.com/nobl/ioBroker.senec/blob/v1.3.9/main.js#L153-L238) lässt sich klar erkennen, dass für Low-Prio (alle Daten) und High-Prio (ausgewählte Daten) jeweils **ein** Request zusammengebaut wird. Warum der Entwickler davon ausgeht, dass ein Abholen aller Daten den Speicher unter große Last setzt und dies daher nur selten gemacht werden sollte, kann ich somit nicht erklären. Vielleicht ist es beim V2.1 so, bei meinem V3 sehe ich das Problem jedenfalls nicht.

**Comment by derheinrich2019** [comment]
Das Dashboard hat mir sofort gefallen und möchte es gerne "sehen" :)

Meine Konfig: Solaredge --> evcc --> solectrus (mqtt-collector)

Ich bin die Anleitung soweit erfolgreich durchgegangen. Einziger Unterschied: Statt einem Pi hab ich eine Proxmox-VM mit arm eingesetzt. Läuft auch alles - es werden leider nicht die Daten aus evcc angezeigt.

Das Docker-Log zeigt mir an, dass die Abfrage der MQTT-Daten klappt:
mqtt-collector-mqtt-collector-1  | 2023-05-24 11:07:01 +0000 {"house_power"=>801}
mqtt-collector-mqtt-collector-1  | 2023-05-24 11:07:09 +0000 {"wallbox_charge_power"=>0}
mqtt-collector-mqtt-collector-1  | 2023-05-24 11:07:09 +0000 {"inverter_power"=>6983}
mqtt-collector-mqtt-collector-1  | 2023-05-24 11:07:09 +0000 {"bat_fuel_charge"=>77.0}

Wenn ich über den Influxdb-Admin die Daten anschaue, sind dort der Forecast und die PV-Daten vorhanden.

Im Dashboard kommen keine Daten an, nur in der Tagesansicht werden die Forecastdaten angezeigt. Daher würde ich vermuten, dass auch die Anbindung influxdb klappt.

Achja, im Docker-Compose-File hab ich den Container für Senec auskommentiert.

Vielleicht hat ja @ledermann eine Idee?

Besten Dank schonmal!

**Comment by ledermann** [comment]
Mittlerweile gibt es den MQTT-Collector in einer experimentellen ersten Fassung:
https://github.com/solectrus/mqtt-collector

Diese Diskussion hier schließe ich nun, da die grundsätzliche Frage ja geklärt ist. Bei konkreten Problemen mit der MQTT-Anbindung bitte ein Issue im zugehörigen Repository anlegen:
https://github.com/solectrus/mqtt-collector/issues

@derheinrich2019 Ich habe für Deine Frage ein Issue angelegt (https://github.com/solectrus/mqtt-collector/issues/23) und antworte dort.
